qemu/tests/cfg/device_assignment.cfg

```diff
@@ -7,7 +7,19 @@
     variants:
         - multi_pf_boot:
             type = boot
-            nics = "nic1 nic2"
+            nics = "nic1 nic2 nic3"
+            pci_assignable_nic1 = no
+            nic_model_nic1 = virtio
+            driver_option = 0
+            variants:
+                - 82576:
+                    pf_filter_re = 82576
+                - XL710:
+                    pf_filter_re = XL710
+                - 82599:
+                    pf_filter_re = 82599
+                - BCM57810:
+                    pf_filter_re = BCM57810
```

Comment:
@balamuruhans have the same concern with you, however, I can't say which is more user friendly. To developers, it's indeed annoying to add variant for every new card. To testers, they do want less manual configuration.